kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/SaslAuthenticator.java

```diff
@@ -0,0 +1,265 @@
+            throw e;
+        }
+
+        try {
+            authState = authenticationProvider.newAuthState(
+                    AuthData.of(saslAuth.getAuthData().getBytes(UTF_8)), null, null);
+        } catch (AuthenticationException e) {
+            responseFuture.complete(new SaslAuthenticateResponse(Errors.SASL_AUTHENTICATION_FAILED, e.getMessage()));
+            return;
+        }
+        authDataSource = authState.getAuthDataSource();
+        authRole = authState.getAuthRole();
+
+        // TODO: we need to let KafkaRequestHandler do the authorization works. Here we just check the permissions
+        //  of the namespace, which is the namespace.
+        final String namespace = saslAuth.getUsername();
+        try {
+            Map<String, Set<AuthAction>> permissions = admin.namespaces().getPermissions(namespace);
+            if (permissions.containsKey(authRole)) {
+                responseFuture.complete(new SaslAuthenticateResponse(Errors.NONE, ""));
+                setState(State.COMPLETE);
+            } else {
+                responseFuture.complete(new SaslAuthenticateResponse(
+                        Errors.SASL_AUTHENTICATION_FAILED,
+                        "Role: " + authRole + " is not allowed on namespace " + namespace));
+            }
+        } catch (PulsarAdminException e) {
+            AuthenticationException authException =
+                    new AuthenticationException("Failed to get permissions of " + namespace + ": " + e.getMessage());
+            responseFuture.complete(
+                    new SaslAuthenticateResponse(Errors.SASL_AUTHENTICATION_FAILED, authException.getMessage()));
+            responseFuture.complete(request.getErrorResponse(authException));
+        }
+    }
+
+    private void handleApiVersionsRequest(ApiVersionsRequest request,
+                                          CompletableFuture<AbstractResponse> responseFuture)
+            throws AuthenticationException {
+        if (state != State.HANDSHAKE_OR_VERSIONS_REQUEST) {
+            throw new IllegalStateException(
+                    "Receive ApiVersions request", state, State.HANDSHAKE_OR_VERSIONS_REQUEST);
+        }
+        if (request.hasUnsupportedRequestVersion()) {
+            responseFuture.complete(request.getErrorResponse(0, Errors.UNSUPPORTED_VERSION.exception()));
+        } else {
+            responseFuture.complete(ApiVersionsResponse.defaultApiVersionsResponse());
+            // Handshake request must be followed by the ApiVersions request
+            setState(State.HANDSHAKE_REQUEST);
+        }
+    }
+
+    private void handleHandshakeRequest(SaslHandshakeRequest request,
+                                        CompletableFuture<AbstractResponse> responseFuture)
+            throws AuthenticationException {
+
+        final String mechanism = request.mechanism();
+        if (mechanism != null && !mechanism.equals("PLAIN")) {
+            // TODO: support more mechanisms
```

Comment:
seems not very clear what to do here, how about open an issue, and describe the detailed requests?